packages/headless/src/controllers/core/did-you-mean/headless-core-did-you-mean.test.ts

```diff
@@ -41,4 +41,12 @@ describe('did you mean', () => {
     dym.applyCorrection();
     expect(engine.actions).toContainEqual(applyDidYouMeanCorrection('bar'));
   });
+
+  it('should allow to disable automatic query correction', () => {
+    engine.state.didYouMean.automaticallyCorrectQuery = false;
+    initDidYouMean();
+
+    dym.applyCorrection();
+    expect(engine.actions).not.toContainEqual(applyDidYouMeanCorrection('bar'));
+  });
```

Comment:
What you're currently testing is that it disables manual query correction, which should still be allowed when automatic query corrections are disabled.

Automatic query correction happens automatically in Headless without calling a method. Specifically, it happens in `packages/headless/src/features/search/search-actions-thunk-processor.ts` when the method `processQueryCorrectionsOrContinue` is called.

Reply:
Yup, my mistake! Redid my changes to the files you suggested and it should work as expected now (tested locally) 🙂 

Thanks for pointing this out!